### PR TITLE
Implement editappt

### PR DESCRIPTION
**Pull request by alfaloo**

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java
Comment (Alteqa) [inline src/main/java/seedu/address/model/ModelManager.java:1]: LGTM

File: src/test/java/seedu/address/logic/commands/EditAppointmentCommandTest.java
Comment (Alteqa) [inline src/test/java/seedu/address/logic/commands/EditAppointmentCommandTest.java:1]: Clean i like it!

File: src/test/java/seedu/address/testutil/EditAppointmentDescriptorBuilder.java
Comment (Alteqa) [inline src/test/java/seedu/address/testutil/EditAppointmentDescriptorBuilder.java:1]: looks right!

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java
Comment (Alteqa) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:1]: nice changes.

**Review by Alteqa — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditAppointmentCommand)) {
+            return false;
+        }
+
+        EditAppointmentCommand otherEditAppointmentCommand = (EditAppointmentCommand) other;
+        return index.equals(otherEditAppointmentCommand.index)
```
Comment (ararchch) [inline src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java:104]: I think can split this into multiple lines to make it more readable and follow the KISS framework

File: src/main/java/seedu/address/logic/parser/EditAppointmentCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    EditAppointmentCommand.MESSAGE_USAGE), pe);
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_DATE);
```
Comment (ararchch) [inline src/main/java/seedu/address/logic/parser/EditAppointmentCommandParser.java:36]: Method is a little long although not a major issue

**Review by ararchch — APPROVED** [review]

**Pull request merged**